ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v3/fileinput/clientlibs/site/js/fileinputwidget.js

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright 2022 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+/**
+ * This class is responsible for interacting with the file input widget. It implements the file preview,
+ * file list, handling invalid file size, file name, file mime type functionality
+ */
+
+if (typeof window.FileInputWidget === 'undefined') {
+    window.FileInputWidget = class extends FormView.FormFileInputWidget {
+
+        constructor(params) {
+            super(params);
+        }
+
+        attachEventHandlers(widget, dragArea, model) {
+            super.attachEventHandlers(widget, model)
+            const customBtn = dragArea.querySelector(".cmp-adaptiveform-fileinput__widgetlabel")
+            dragArea?.addEventListener("dragover", (event)=>{
+                event.preventDefault();
+                dragArea.classList.add("cmp-adaptiveform-fileinput__dragarea--active");
+            });
+            dragArea?.addEventListener("dragleave", (event)=>{
+                dragArea.classList.remove("cmp-adaptiveform-fileinput__dragarea--active");
+            });
+            dragArea?.addEventListener("drop", (event)=>{
+            event.preventDefault(); 
+            dragArea.classList.remove("cmp-adaptiveform-fileinput__dragarea--active");
+            this.handleChange(event?.dataTransfer?.files);
+            });
+            dragArea?.addEventListener("paste", (event)=>{
+                event.preventDefault();
+                this.handleChange(event?.clipboardData?.files);
+            });
+            customBtn.addEventListener("click", (event) => {
+                event.preventDefault();
+                widget.click();
+            });
+            customBtn.addEventListener('keypress', function(event) {
+                // Check if the Enter key is pressed
+                if (event.key === 'Enter' || event.keyCode === 13) {
+                    // Trigger the click event of the file input
+                    event.preventDefault();
+                    widget.click();
+                }
+            });
+        }
+        'hello'
+        fileItem(fileName, fileSize, comment, fileUrl) {
```

Comment:
Similarly call the super method first and only add the code specific to v3

Reply:
As discussed, I will pick this as my next task as it requires some time to refactor the code.